### PR TITLE
Remove tracker and tracker_worker

### PR DESCRIPTION
**Pull request by sondreso**
**Issue**
Resolves #my_issue

**Approach**
_Short description of the approach_

(Screenshot of new behavior in GUI if applicable)

## Pre review checklist

- [ ] Read through the code changes carefully after finishing work
- [ ] Make sure tests pass locally (after every commit!)
- [ ] Prepare changes in small commits for more convenient review (optional)
- [ ] PR title captures the intent of the changes, and is fitting for release notes.
- [ ] Updated documentation
- [ ] Ensured that unit tests are added for all new behavior (See
    [Ground Rules](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md#ground-rules)),
    and changes to existing code have good test coverage.

## Pre merge checklist
- [ ] Added appropriate release note label
- [ ] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).

### CODE REVIEW

**Comment by oyvindeide** [comment]
A very non-scientific test, but `test_cli.py` was 30% faster in this branch vs main

File: src/ert/cli/main.py [hunk 1]
```diff
@@ -136,12 +135,12 @@ def run_cli(args: Namespace, _: Any = None) -> None:
...
         monitor = Monitor(out=out, color_always=args.color_always)
-
+        thread.start()
         try:
-            monitor.monitor(tracker.track())
+            monitor.monitor(status_queue)
         except (SystemExit, KeyboardInterrupt):
             print("\nKilling simulations...")
-            tracker.request_termination()
+            # tracker.request_termination()
```
Comment (oyvindeide) [inline src/ert/cli/main.py:143]: Can we do anything for cancelling cli?

Comment (sondreso) [reply src/ert/cli/main.py]: Done :heavy_check_mark:

File: src/ert/cli/main.py [hunk 2]
```diff
@@ -130,12 +129,12 @@ def run_cli(args: Namespace, _: Any = None) -> None:
...
-
+        thread.start()
         try:
-            monitor.monitor(tracker.track())
-        except (SystemExit, KeyboardInterrupt):
+            monitor.monitor(status_queue)
+        except (SystemExit, KeyboardInterrupt, OSError):
             print("\nKilling simulations...")
-            tracker.request_termination()
+            model.cancel()
```
Comment (oyvindeide) [inline src/ert/cli/main.py:137]: Also need to stop the monitor here?

File: src/ert/cli/monitor.py [hunk 3]
```diff
@@ -57,13 +58,15 @@ def __init__(self, out: TextIO = sys.stdout, color_always: bool = False) -> None
...
+        self.done = False
 
     def monitor(
         self,
-        events: Iterator[Union[FullSnapshotEvent, SnapshotUpdateEvent, EndEvent]],
+        event_queue: SimpleQueue,
     ) -> None:
         self._start_time = datetime.now()
-        for event in events:
+        while True:
```
Comment (sondreso) [inline src/ert/cli/monitor.py:68]: This is a bit dangerous

File: src/ert/cli/main.py (hunk 2 above)
Comment (sondreso) [reply src/ert/cli/main.py:130]: We can make it explicit that this communicates over a queue on the outside, but I think the next step here is the simplify the entire run model API anyway?

File: src/ert/run_models/base_run_model.py [hunk 4]
```diff
@@ -373,18 +421,159 @@ def checkHaveSufficientRealizations(
...
+                )
+            )
+
     def run_ensemble_evaluator(
         self, run_context: RunContext, ee_config: EvaluatorServerConfig
     ) -> List[int]:
+        if not self._end_queue.empty():
+            event_logger.debug("Run model canceled - pre evaluation")
+            self._end_queue.get()
+            return []
```
Comment (oyvindeide) [inline src/ert/run_models/base_run_model.py:504]: Should this be a separate function we call periodically?

**Review by oyvindeide — APPROVED** [review]
LGTM! Really nice improvement 🎖️

File: src/ert/run_models/base_run_model.py (hunk 4 above)
Comment (sondreso) [reply src/ert/run_models/base_run_model.py:504]: Probably at a later point yes, we want it also in the analysis update :thinking:

**Pull request merged**